runtimes/peregrine/src/xcm_config.rs

```diff
@@ -160,16 +164,40 @@ impl Contains<RuntimeCall> for SafeCallFilter {
 	}
 }
 
+parameter_types! {
+	pub TreasuryAccountId: AccountId = Treasury::account_id();
+}
+
 pub struct XcmConfig;
 impl xcm_executor::Config for XcmConfig {
 	type RuntimeCall = RuntimeCall;
 	// How we send Xcm messages.
 	type XcmSender = XcmRouter;
 	// How to withdraw and deposit an asset.
-	type AssetTransactor = LocalAssetTransactor<Balances, RelayNetworkId>;
+	// Until fixed, `LocalAssetTransactor` must be last since it returns an error if
+	// the operation does not go through, i.e., it cannot be chained with other
+	// transactors.
+	type AssetTransactor = (
+		// Allow the asset from the other side of the pool to be "deposited" into the current system.
+		SwapPairRemoteAssetTransactor<LocationToAccountIdConverter, Runtime>,
+		// Allow the asset to pay for remote XCM fees to be deposited into the current system.
+		FungiblesAdapter<
+			Fungibles,
+			MatchesSwapPairXcmFeeFungibleAsset<Runtime>,
+			LocationToAccountIdConverter,
+			AccountId,
+			NoChecking,
+			CheckingAccount,
+		>,
+		// Transactor for fungibles matching the "Here" location.
+		LocalAssetTransactor<Balances, RelayNetworkId>,
+	);
```

Comment:
Yes, the same as above applies. I'll move these into the common crate before the final review.

Reply:
Actually for now, since they're not in Spiritnet, I would keep them here. I will move them into `runtime-common` once we add the same stuff to Spiritnet, and it's easier to see what can be factored out into a common configuration.

Reply:
Also, since a few other PRs are based on top of this, including a renaming one, it just makes it easier to merge, for now.